### PR TITLE
Support thrift binary non-ssl and ssl connections at the same time

### PR DESCRIPTION
**Pull request by turboFei**
### _Why are the changes needed?_

Before we only support thrift binary non-ssl, since #3658, we support thrift binary ssl connection.

If customer want to migrate from non-ssl to ssl thrift binary connections, we need support non-ssl and ssl connections at the same time.

This PR will help for the thrift binary ssl migration.

For example:

before: 
```
kyuubi.frontend.protocols=THRIFT_BINARY
kyuubi.frontend.thrift.binary.bind.port=10009
```

after:
```
kyuubi.frontend.protocols=THRIFT_BINARY,THRIFT_BINARY_SSL
kyuubi.frontend.thrift.binary.bind.port=10009
kyuubi.frontend.thrift.binary.ssl.bind.port=10010
```
### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [x] Add screenshots for manual tests if appropriate
image

- [x] [Run test](https://kyuubi.readthedocs.io/en/master/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by turboFei** [comment]
how do you think about? @yaooqinn  @pan3793 

I think it should be common case if they want to make SSL migration.

File: docs/deployment/settings.md [hunk 1]
```diff
@@ -311,6 +311,7 @@ You can configure the Kyuubi properties in `$KYUUBI_HOME/conf/kyuubi-defaults.co
...
+| kyuubi.frontend.thrift.binary.ssl.bind.port            | 10009             | Port of the machine on which to run the thrift frontend service via the binary protocol with SSL enabled.                                                                                                                                                                                                                                                                                                                                                                                                                   | int      | 1.7.0 |
```
Comment (pan3793) [inline docs/deployment/settings.md:314]: we should give it a different value

**Comment by pan3793** [comment]
I think it's a common case, many services support to listen plain/ssl port at the same time.

**Review by bowenliang123 — APPROVED** [review]
LGTM

**Review by pan3793 — APPROVED** [review]

**Comment by yaooqinn** [comment]
A bit weird to me for the naming part

**Comment by yaooqinn** [comment]
This does not support HA and introduces bugs when the SSL on/off together w/ the same namespace.

File: docs/deployment/settings.md [hunk 2]
```diff
@@ -311,6 +311,7 @@ You can configure the Kyuubi properties in `$KYUUBI_HOME/conf/kyuubi-defaults.co
...
+| kyuubi.frontend.thrift.binary.ssl.bind.port            | 10011             | Port of the machine on which to run the thrift frontend service via the binary protocol with SSL enabled.                                                                                                                                                                                                                                                                                                                                                                                                                   | int      | 1.7.0 |
```
Comment (yaooqinn) [inline docs/deployment/settings.md:314]: kyuubi.frontend.thrift.binary.bind.port also can be used for ssl, how do we suppose users to understand?

File: kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala [hunk 3]
```diff
@@ -371,7 +371,7 @@ object KyuubiConf {
 
   object FrontendProtocols extends Enumeration {
     type FrontendProtocol = Value
-    val THRIFT_BINARY, THRIFT_HTTP, REST, MYSQL, TRINO = Value
+    val THRIFT_BINARY, THRIFT_BINARY_SSL, THRIFT_HTTP, REST, MYSQL, TRINO = Value
```
Comment (yaooqinn) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala:374]: THRIFT_BINARY, THRIFT_BINARY_SSL seems not discrete mathematically

Comment (pan3793) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala:374]: I think the idea here is:

1. THRIFT_BINARY, THRIFT_BINARY_SSL should be like HTTP, HTTPS eventually, there is discrete and can be enabled at the same time.
2. currently we keep SSL option of THRIFT_BINARY for compatibility, the deprecating message has been added and this feature will be removed in the 2.0

**Pull request closed without merging**